teos/src/api/internal.rs

```diff
@@ -280,27 +279,40 @@ impl PrivateTowerServices for Arc<InternalAPI> {
                 .map_or("an unknown address".to_owned(), |a| a.to_string())
         );
 
-        let mut matching_appointments = vec![];
-        let locator = Locator::from_slice(&request.into_inner().locator).map_err(|_| {
+        let req_data = request.into_inner();
+        let locator = Locator::from_slice(&req_data.locator).map_err(|_| {
             Status::new(
                 Code::InvalidArgument,
                 "The provided locator does not match the expected format (16-byte hexadecimal string)",
             )
         })?;
 
-        for (_, appointment) in self
+        let mut appointments: Vec<(UserId, Appointment)> = self
             .watcher
             .get_watcher_appointments_with_locator(locator)
+            .into_values()
+            .map(|appointment| (appointment.user_id, appointment.inner))
+            .collect();
+
+        let user_id_slice = req_data.user_id;
+        if !(&user_id_slice.is_empty()) {
```

Comment:
This is not how you should approach this, right no the "optionality" of the server side comes form the slice being empty. That is not a good practice. Do make the proto field optional so the compiled object here is an `Option<T>` and you can do the proper checks with it

Reply:
Oh yeah, totally missed that